ci/do_ci.sh

```diff
@@ -89,6 +89,7 @@ if [[ "$1" == "cmake.test" ]]; then
   cd "${BUILD_DIR}"
   rm -rf *
   cmake -DCMAKE_BUILD_TYPE=Debug  \
+        -DCMAKE_CXX_STANDARD=14 \
```

Comment:
This property would be redundant once we switch to C++11 (#2147), as we would update the property [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/ca8c23481ae10c1569858cee19fa076c27b8f261/CMakeLists.txt#L139). Need to make a note of removing from here once that is done.

Reply:
Added a note in #2147, so this part is resolved (nothing to change in this PR).